source/common/http/header_map_impl.cc

```diff
@@ -91,18 +109,17 @@ void HeaderString::append(const char* data, uint32_t size) {
     // We can get here either because we didn't fit in inline or we are already dynamic.
     if (type_ == Type::Inline) {
       const uint64_t new_capacity = (static_cast<uint64_t>(string_length_) + size) * 2;
-      // If the resizing will cause buffer overflow due to hitting uint32_t::max, an OOM is likely
```

Comment:
Do you want to change `new_capacity` above this to be `size_t`? Other than that, LGTM.